engines/saga/image.cpp

```diff
@@ -96,26 +114,49 @@ bool SagaEngine::decodeBGImage(const ByteArray &imageData, ByteArray &outputBuff
 	readS.readUint16();
 	readS.readUint16();
 
-	outputBuffer.resize(hdr.width * hdr.height);
-
 	if (isAGA() || isECS()) {
-		int planePitch = (hdr.width + 15) & ~15;
-		int linePitch = isAGA() ? planePitch : (planePitch * 5 / 8);
 		unsigned bitnum = isAGA() ? 8 : 5;
 		int headerSize = 8 + (3 << bitnum);
 		const byte *RLE_data_ptr = &imageData.front() + headerSize;
 		size_t RLE_data_len = imageData.size() - headerSize;
 
-		if (RLE_data_len != (size_t) linePitch * hdr.height)
-			return false;
-
-		memset(outputBuffer.getBuffer(), 0, hdr.width * hdr.height);
-		for (int y = 0; y < hdr.height; y++)
-			for (int x = 0; x < hdr.width; x++)
-				for (unsigned bit = 0; bit < bitnum; bit++) {
-					int inXbit = x + bit * planePitch;
-					outputBuffer[y * hdr.width + x] |= ((RLE_data_ptr[y * linePitch + inXbit / 8] >> (7 - inXbit % 8)) & 1) << bit;
-				}
+		if (getFeatures() & GF_POWERPACK_GFX) {
```

Comment:
As mentioned in the game entries, we could avoid introducing another game feature and use `getPlatform()` here instead:
```suggestion
		if (_vm->getPlatform() == Common::kPlatformAmiga) {
```

Reply:
Nope, doesn't work